scio-core/src/main/scala/com/spotify/scio/coders/Coder.scala

```diff
@@ -318,33 +293,29 @@ final private[scio] case class RecordCoder[T](
     }
 
   override def encode(value: T, os: OutputStream): Unit = {
-    var i = 0
-    val array = destruct(value)
-    while (i < array.length) {
-      val (label, c) = cs(i)
-      val v = array(i)
-      onErrorMsg(
-        s"Exception while trying to `encode` an instance of $typeName:  Can't encode field $label value $v"
-      ) {
-        c.encode(v, os)
+    destruct(value)
+      .zip(cs)
+      .foreach { case (v, (l, c)) =>
```

Comment:
Not sure if working with arrays was done fore performance reasons

Reply:
Yep